### PR TITLE
first pass at project onboarding checklist per #121

### PR DESCRIPTION
**Pull request by jorydotcom**
Per our meeting today, this takes the initial tasks we identified in #121 and organizes them a bit as a checklist proposal. 

Reviews requested!

### CODE REVIEW

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Onboarding Checklist
+
+This is an informational checklist to help projects onboard into the OpenJS Foundation. If you have any questions or need help, the OpenJS Foundation CPC is available to assist.
+
+- [ ] Adopt the OpenJS Foundation Code of Conduct
```
Comment (ljharb) [inline proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Is this a requirement prior to joining, or is a commitment to do so sufficient? I have a project that i very much want to join and use this CoC, but prior to living in a neutral home i may be unable to easily make these sorts of changes.

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+- [ ] Update project CoC reporting methods to include OpenJS Foundation escalation path
+- [ ] Transfer official domains to OpenJS Foundation
```
Comment (ljharb) [inline proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Similarly, this feels like putting the cart before the horse; i should forfeit ownership of assets prior to obtaining the membership which preserves my access to those assets?

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- [ ] Identify and document other core project infrastructure
+- [ ] Choose either CLA or DCO; implement appropriate tool
```
Comment (ljharb) [inline proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: is a CLA really required? https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/ seems convincing to me

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md (hunk 1 above)
Comment (mhdawson) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: This is a checklist for projects joining.  Not things that need to be done in advance.  I believe a commitment to do so, and then doing it with X time after joining is what we had in mind.

Comment (ljharb) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Perfect, that clarification helps :-)

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md (hunk 3 above)
Comment (jorydotcom) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: No, not at all - projects can chose what works for them.

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md (hunk 2 above)
Comment (jorydotcom) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Definitely keep your ownership prior to becoming a member; as you onboard we'll figure out what timeframe makes sense to transition assets over once you've become an OpenJS Foundation project

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md (hunk 3 above)
Comment (ljharb) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Thanks, that makes sense. The wording here (and the "checklist" form) implied to me that all were required.

Comment (jorydotcom) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: Good feedback, I'll write some clarifying copy here!

File: proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md (hunk 1 above)
Comment (jorydotcom) [reply proposals/stage-1/PROJECT_ONBOARDING/PROJECT_ONBOARDING.md]: added some clarifying prose!

**Comment by mhdawson** [comment]
Updates look good to me:)

**Pull request merged**